coderd/database/dbmem/dbmem.go

```diff
@@ -9498,6 +9498,24 @@ func (q *FakeQuerier) ListWorkspaceAgentPortShares(_ context.Context, workspaceI
 	return shares, nil
 }
 
+func (q *FakeQuerier) MarkAllInboxNotificationsAsRead(_ context.Context, arg database.MarkAllInboxNotificationsAsReadParams) error {
+	err := validateDatabaseType(arg)
+	if err != nil {
+		return err
+	}
+
+	for idx, notif := range q.inboxNotifications {
+		if notif.UserID == arg.UserID && !notif.ReadAt.Valid {
+			q.inboxNotifications[idx].ReadAt = sql.NullTime{
+				Time:  dbtime.Now(),
+				Valid: true,
```

Comment:
Shouldn't you use `ReadAt` from `arg`?

Reply:
Not sure if you have a use case in mind - having it as an arg would mean the user can set the notification as read 'in the past'.

Reply:
Take a look at `coderd/inboxnotifications.go`:

```
err := api.Database.MarkAllInboxNotificationsAsRead(ctx, database.MarkAllInboxNotificationsAsReadParams{
		UserID: apikey.UserID,
		ReadAt: sql.NullTime{Time: dbtime.Now(), Valid: true},
	})
```

This is precisely how you're calling it. You need to go one way or another:
1. Remove `ReadAt` from the arg.
2. Modify dbmem to take `dbtime.Now()` from coderd.

Let me know if this is clear. I can hop on zoom to explain.